immuni_analytics/helpers/device_check.py

```diff
@@ -0,0 +1,137 @@
+#   Copyright (C) 2020 Presidenza del Consiglio dei Ministri.
+#   Please refer to the AUTHORS file for more information.
+#   This program is free software: you can redistribute it and/or modify
+#   it under the terms of the GNU Affero General Public License as
+#   published by the Free Software Foundation, either version 3 of the
+#   License, or (at your option) any later version.
+#   This program is distributed in the hope that it will be useful,
+#   but WITHOUT ANY WARRANTY; without even the implied warranty of
+#   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+#   GNU Affero General Public License for more details.
+#   You should have received a copy of the GNU Affero General Public License
+#   along with this program. If not, see <https://www.gnu.org/licenses/>.
+
+import logging
+import uuid
+from datetime import datetime
+from json import JSONDecodeError
+from typing import Any, Dict
+
+import jwt
+from aiohttp import ClientError, ClientSession
+
+from immuni_analytics.core import config
+from immuni_analytics.helpers.request import (
+    BadFormatRequestError,
+    ServerUnavailableError,
+    post_with_retry,
+)
+from immuni_analytics.models.device_check import DeviceCheckData
+
+_LOGGER = logging.getLogger(__name__)
+
+_DEVICE_CHEK_URL = "https://api.development.devicecheck.apple.com/v1"
+_DEVICE_CHECK_GET_BITS_URL = f"{_DEVICE_CHEK_URL}/query_two_bits"
+_DEVICE_CHECK_SET_BITS_URL = f"{_DEVICE_CHEK_URL}/update_two_bits"
+
+
+class DeviceCheckApiError(Exception):
+    """Raised when an error occurs while calling the DeviceCheck API"""
```

Comment:
I'll comment only this docstring, yet please double-check the whole module for trailing `.`s.